packages/astro/src/runtime/server/render/page.ts

```diff
@@ -106,6 +122,14 @@ export async function renderPage(
 							}
 							controller.close();
 						} catch (e) {
+							// We don't have a lot of information downstream, and upstream we can't catch the error properly
+							// So let's add the location here
+							if (AstroError.is(e) && !e.loc) {
```

Comment:
It's a bit dirty to do here, but due to how readablestreams works, we can't catch it higher than that (and downstream we don't have the route)